blaze/compute/core.py

```diff
@@ -446,10 +446,9 @@ def compute(expr, d, **kwargs):
     optimize_ = kwargs.get('optimize', optimize)
     pre_compute_ = kwargs.get('pre_compute', pre_compute)
     post_compute_ = kwargs.get('post_compute', post_compute)
-
     expr2, d2 = swap_resources_into_scope(expr, d)
     if pre_compute_:
-        d3 = dict([(e, pre_compute_(expr2, dat, **kwargs))
+        d3 = dict([(e, pre_compute_(e, dat, **kwargs))
```

Comment:
it looks like your fix is correct. before your fix the code is mapping each subexpression to the same `pre_compute` result. after your fix each subexpression in `expr2` gets mapped to the correct `pre_compute`d result.